content/index.md

```diff
@@ -0,0 +1,172 @@
+
+> As a developer, I am creating an application that sends and receives SMS, using the Messages API for outgoing messages.
+
+> As a developer, I need to inspect the configuration of my account and may also want to change one or more settings.
+
+> As a developer, I haven't used Nexmo since I first looked at it a few weeks ago. I want to look at which applications I have and which numbers are used by which application so that I can decide if I can reallocate or cancel numbers.
+
+> As a developer I want to look up some information about a user's number for security purposes.
+
+> I have been asked to detail for our auditors all the numbers on our Nexmo account and which applications use them. I need to supply this as a spreadsheet.
+
+> As a developer, we run a large number of applications, and each one may have multiple incoming numbers linked to it. I want to know which numbers are linked to application X.
+
+
+## Packaging, Installation and Setup
+
+CLI tools **must**:
+
+ - be available on all platforms (Windows, Mac, Linux - any others would be a bonus)
+ - have as few dependencies as possible
+ - be open to users of all our supported technology stacks (Ruby, PHP, Python, Java, NodeJS, .NET, Go, Android, iOS)
+
+In addition, it would be nice if the tool could:
+
+ - upgrade itself
+ - be extensible with plugins
+ - be relatively small in size
+ - not store user's actual credentials on disk
+
+The tool will be able to state its current version when `--version` is supplied
+
+## Command Design
+
+Consistency and predictability are the key to a command line tools that users become familiar and comfortable with. To this end, all commands will follow the same conventions to enable us to build and maintain a tool that continues to feel familiar.
+
+```
+awesomecli topic:command --kebab-case-param value 
+```
+
+The above shows a basic template to use as a starting point
+
+* **`topic`** will usually be a plural noun. Examples could include `applications`, `numbers`, `insights`.
+* **`command`** will usually be an active or imperative verb. Examples could include `create`, `list`, `configure`.
+* For more complex areas, add a **subtopic** with the format `topic:subtopic:command`. Examples could include `accounts:jwt:generate`, `applications:numbers:link`.
+* All input will be supplied with clearly-named flags each followed by the value for this parameter. The parameters will use "kebab case" and will be named exactly the same when different commands accept the same input data. Examples could include `--number`, `--application`, `--event_url`.
+* If a command can only accept one input parameter, it may be supplied without flags, as an additional shorthand to the standard named-flag approach.
+
+The tool will support autocomplete so verbose is better. 
+
+## Interactive Commands
+
+**All commands must be available in non-interactive mode** to enable the tool to be used in automation.
```

Comment:
would interactive mode be default, with a per command flag to make it non-interactive? or would that be a setup flag?

Reply:
I'd say interactive mode is the default if you miss a mandatory parameter. Then add the ability to raise an error rather than going interactive via `.nexmorc`

Reply:
I would suggest having interactive mode _not_ be the default. I can't think of many command-line tools that default to interactive behaviour.

Reply:
I was sort of thinking to go interactive if the user forgot a required flag, to prompt them. I'm not sure the interactive mode has value otherwise, because once we've jarred them with an error, they might as well go read the help at that point? I'm happy to drop the interactive if it's not seen as useful.

Reply:
I *personally* wouldn't find it particularly useful (I may be an outlier though). In my experience, returning a usage summary and an error code and allowing the user to amend their command is as useful, or maybe more useful (because the completed command becomes something they can copy-paste for whatever reason they might have).

Reply:
I've had the opposite experience. I'm very pro-scripting, but working with the Azure CLI tooling it dropped in to interactive mode if you don't provide a required field and it was great for exploring how to use the tool